core/src/main/java/com/equo/application/client/EquoApp.java

```diff
@@ -96,12 +94,13 @@ public static void setAppName(String appName) {
     var appId = appName.toLowerCase().replaceAll("[^a-z0-9 -]", "").replaceAll(" +", " ").trim()
         .replaceAll(" ", "-");
     if (!appId.isEmpty()) {
-      appId = appId.substring(0, Math.min(50, appId.length() - 1));
+      appId = appId.substring(0, Math.min(50, appId.length()));
       if (appId.charAt(appId.length() - 1) == '-') {
         appId = appId.substring(0, appId.length() - 1);
       }
```

Comment:
Since we're modifying this function, we could review if the appId meets certain conditions: it should not start or end with a hyphen "-" and should not have two consecutive hyphens "-".

Reply:
We would need to add the corresponding tests.